_posts/en/newsletters/2018-12-18-newsletter.md

```diff
@@ -0,0 +1,200 @@
+    announcements, plus overhead, for what is (on average) just a 400
+    byte transaction.  With sketches, you should be able to send small
+    periodic sketches to your peers and they will attempt to reconstruct
+    any transactions they're missing.  If you don't send enough data
+    to them to reconstruct some transactions, they can try again after
+    receiving additional sketches either from you or from their other
+    peers.  An [estimate][nmnkgl relay] based on earlier versions of
+    Bitcoin Core shows that significantly improved relay efficiency
+    could reduce the amount of bandwidth being wasted by long-lived
+    nodes by up to 90%.  Sketches also have the potential to provide
+    other desirable features---for example, LN protocol developer Rusty
+    Russel started a [thread][ln minisketch] on the Lightning-Dev
+    mailing list about using them for sending LN routing table updates.
+
+- **Users planning withdrawals from exchanges January 3rd:** several
+  people popular on Bitcoin Twitter have announced their intention to
+  withdraw all their bitcoins from exchanges for the day of Thursday,
+  January 3rd, and an unknown number of less well known users are likely
+  to do the same.  One of their stated goals is withdraw enough bitcoins
+  from exchanges to publicly reveal whether any exchanges are insolvent.
+
+    Exchanges should prepare for how they can handle an increased number
+    of withdrawals---you do not want to be tempted into overriding your
+    usual cold storage withdrawal policies because some users are
+    accusing you of being insolvent.  Users should ensure that they have
+    the ability to safely withdraw and store their bitcoins before they
+    attempt to transfer their funds---if you fail to do so properly, you
+    may lose all of you bitcoins.  Everyone should also be prepared for
+    higher transaction fees that day due to increased onchain demand.
+
+    We also wish to remind both users and exchanges that there are known
+    protocols ([1][maxwell-todd por], [2][boneh-et-al por]) that allow
+    exchanges to provide users with cryptographic proof of reserves.
+    These are superior to temporary bitcoin withdrawal ceremonies in all
+    ways: they can be performed by the exchange on a regular basis with
+    no user effort, they don't require withdrawing 100% of deposits from
+    an exchange to prove it can sign for 100% of reserves, and they
+    don't bloat the block chain or increase transaction fees.  Optech
+    suggests future initiatives focus on users requesting (and exchanges
+    providing) proof of reserves.
+
+- **Description about what might be included in a Schnorr/Taproot soft
+  fork:** Bitcoin protocol developer Anthony Towns has [posted][towns
+  schnorr taproot] a quite readable email describing what he thinks
+  ought to be included in a soft fork that adds the Schnorr signature
+  scheme plus Taproot-style MAST to Bitcoin.  These are just his
```

Comment:
I'd avoid the word MAST in favour of *Merkle branches in script* or similar. See https://www.youtube.com/watch?v=YSUVRj8iznU&feature=youtu.be&t=1377: "Merklized Abstract Syntax Trees as it was originally called but that's at least according to its original author Russell O'Connor not what we should be talking about when we talk about Merkle branches today."

Reply:
I'm fine to leave MAST in this newsletter (after all, the author gets to choose the style), but my guess is that when a soft fork proposal is made, it won't include the word MAST. AJ uses *taproot/merkle path* in his email and not MAST, and Pieter uses the term *Merkle branches in script* in his slide.

Rambly aside: I remember being confused by the term *MAST* when I first encountered it. In Bitcoin, MAST refers to a tree where each leaf is a disjunctive condition (A or B or C or ... must hold for the whole tree to evaluate to True). In an Abstract Syntax Tree, the leaves are not disjoint, so intermediate nodes can represent conjunctive conditions. My assumption when first seeing the term MAST was that opcodes like OP_AND could occur at the intermediate nodes, rather than just OP_IF, and that Merkle paths would be given for all the leaf nodes that needed to be evaluated. Obviously nonsense, but that's what the term initially suggested to me! I think *Merkle branches in script* or *Merkle script tree* or similar is more intuitive.

Reply:
> I think Merkle branches in script or Merkle script tree or similar is more intuitive.

Fair enough.  From my perspective of never having taken a CS course, none of them (nor MAST) give me any intuitive understanding about how they work or what benefits they provide.  Indeed, if you dropped any of those terms into an episode of Star Trek or Dr. Who, I doubt I'd have noticed them amongst the other technobable.  :-)

I rather like AJ's [backronym](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2018-November/016500.html) of the name.  It isn't any more intuitive, but it avoids the confusion that you describe above and allows us to continue to use the well-established (and concise!) term.

> I think I'm going to claim "MAST" stands for "merkelized alternative script tree" these days, since they're not "abstract syntax trees"